cli/src/cmd/mock-protected-data.js

```diff
@@ -0,0 +1,226 @@
+import { mkdir, readFile, stat, writeFile } from 'node:fs/promises';
+import { join } from 'node:path';
+import chalk from 'chalk';
+import boxen from 'boxen';
+import { getSpinner } from '../cli-helpers/spinner.js';
+import { fileExists } from '../utils/fs.utils.js';
+import { PROTECTED_DATA_MOCK_DIR } from '../config/config.js';
+import { handleCliError } from '../cli-helpers/handleCliError.js';
+import {
+  createZipFromObject,
+  extractDataSchema,
+  ALLOWED_KEY_NAMES_REGEXP,
+} from '../libs/dataprotector.js';
+
+export async function mockProtectedData() {
+  const spinner = getSpinner();
+
+  try {
+    async function buildData({ dataState = {}, dataSchema = {} } = {}) {
+      // get data fragment key
+      const { key } = await spinner.prompt({
+        type: 'text',
+        name: 'key',
+        message:
+          "What key do you want to use to store the piece of data? (use '.' to access nested keys)",
+      });
+
+      // check key is valid
+      const keyPath = key.split('.');
+      const keyFragmentErrors = keyPath
+        .map((fragment) => {
+          if (fragment === '') {
+            return `Empty key fragment`;
+          }
+          if (!ALLOWED_KEY_NAMES_REGEXP.test(fragment)) {
+            return `Unsupported special character in key`;
+          }
+        })
+        .filter(Boolean);
+
+      // get data fragment type
+      if (keyFragmentErrors.length === 0) {
+        const BOOLEAN = 'boolean';
+        const NUMBER = 'number';
+        const STRING = 'string';
+        const FILE = 'file';
```

Comment:
may be we can use an object here 
```
const DataType = {
  BOOLEAN: 'boolean',
  NUMBER: 'number',
  STRING: 'string',
  FILE: 'file',
};
```

Reply:
I would not, these are internal implementation details, other modules don't care about these values.